lib/client.js

```diff
@@ -389,25 +422,36 @@ Client.prototype.query = function (config, values, callback) {
     query._result._getTypeParser = this._types.getTypeParser.bind(this._types)
   }
 
+  if (!this._queryable) {
+    query.handleError(new Error('Client has encountered a connection error and is not queryable'), this.connection)
+    return
```

Comment:
Returning `undefined` causes breakage for Promise users when calling `.query()` after the connection has gone down. Should return `result` instead.

Reply:
Thanks!